packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_headers_keyboard_navigation.ts

```diff
@@ -113,6 +123,75 @@ export class HeadersKeyboardNavigationController extends ColumnKeyboardNavigatio
     );
   }
 
+  protected getContainerBoundingRect($container: dxElementWrapper) {
+    const containerRect = getBoundingRect($container.get(0));
+
+    return {
+      left: containerRect.left,
+      right: containerRect.right,
+    };
+  }
+
+  protected getScrollPadding(
+    $container: dxElementWrapper,
+  ): {
+      left: number;
+      right: number;
+    } {
+    const containerRect = getBoundingRect($container.get(0));
+    const containerBoundingRect = this.getContainerBoundingRect($container);
+
+    return {
+      left: containerBoundingRect.left - containerRect.left,
+      right: containerRect.right - containerBoundingRect.right,
+    };
+  }
+
+  protected isOutsideVisibleArea = (
+    $element: dxElementWrapper,
+    $container: dxElementWrapper,
+  ): boolean => {
+    const elementRect = getBoundingRect($element.get(0));
+    const elementRectLeft = Math.round(elementRect.left);
+    const elementRectRight = Math.round(elementRect.right);
+    const containerBoundingRect = this.getContainerBoundingRect($container);
+
+    return elementRectLeft < containerBoundingRect.left
+      || elementRectRight > containerBoundingRect.right;
+  };
+
+  protected scrollToColumn($cell: dxElementWrapper): void {
+    const scrollable = this.getView('rowsView')?.getScrollable();
+
+    if (scrollable) {
+      const cellIsOutsideVisibleArea = this.isOutsideVisibleArea(
```

Comment:
Let's rename this variable:
```suggestion
      const isCellOutsideVisibleArea = this.isOutsideVisibleArea(
```